tests/bug-detectors/ssrf.test.js

```diff
@@ -0,0 +1,372 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const dgram = require("dgram");
+const fs = require("fs");
+const http = require("http");
+const https = require("https");
+const path = require("path");
+
+const {
+	FuzzTestBuilder,
+	FuzzingExitCode,
+	JestRegressionExitCode,
+} = require("../helpers.js");
+
+const {
+	host,
+	hostIPv6,
+	okPort,
+	okPortHttps,
+	okPortIPv6,
+	notOkPort,
+	notOkPortHttps,
+	okMessage,
+	notOkMessage,
+} = require("./ssrf/connection-settings.js");
+
+describe("SSRF", () => {
+	let endpoints;
+
+	beforeAll(async () => {
+		endpoints = [
```

Comment:
Do we really need to start up a server, as trying to connect should be enough to trigger the bug detector, right?